packages/grafana-data/src/transformations/matchers/readOnlyMatcher.ts

```diff
@@ -0,0 +1,33 @@
+import { FieldMatcher } from '../../types';
+import { fieldMatchers } from '../matchers';
+import { FieldMatcherID } from './ids';
+
+export interface ReadOnlyFieldMatcherOptions<T = any> {
+  innerId: FieldMatcherID;
+  innerOptions: T;
+  formattedValue: string;
+}
+
+const defaultMatcher: FieldMatcher = () => false;
+
+fieldMatchers.register({
+  id: FieldMatcherID.readOnly,
+  name: 'Read-only field matcher',
+  description: 'Field name by inner matcher',
+  get: (options: ReadOnlyFieldMatcherOptions): FieldMatcher => {
+    if (options.innerId === FieldMatcherID.readOnly) {
+      return defaultMatcher;
+    }
```

Comment:
In such case there is something definitely wrong - what about throwing an error rather than returning default matcher?